src/utils/utils.js

```diff
@@ -1850,6 +1850,43 @@ export function getRangePosition(el, document, options = {}) {
 
     return offset;
 }
-export function getClosestNotEditable(node, root) {
-    return node && ancestors(node, root).includes(root) && node.closest('[contenteditable=false]');
+
+/**
+ * Get the highest not editable node if the `node` is within `root` and is a
+ * non editable node.
+ *
+ * Otherwise return `undefined`.
+ *
+ * Example:
+ *
+ * ```html
+ * <div class="root" contenteditable="true">
+ *     <div class="A">
+ *         <div class="B" contenteditable="false">
+ *             <div class="C">
+ *             </div>
+ *         </div>
+ *     </div>
+ * </div>
+ * ```
+ *
+ * ```js
+ * getHighestNotEditableFromNotEditable(document.querySelector(".C")) // return "B"
+ * ```
+ */
+export function getHighestNotEditableFromNotEditable(node, root) {
+    let currentNode = node;
+    let lastEditable;
+    if (!ancestors(node, root).includes(root)) {
+        return;
+    }
+    while (currentNode && currentNode !== root) {
+        if (currentNode.isContentEditable) {
+            return lastEditable;
+        } else if (currentNode.isContentEditable === false) {
```

Comment:
Why `else if` and not just `else` ? Can `isContentEditable` be `undefined` ? Does your function still work in that case ? Just making sure :-)